src/Aks/Aks/Commands/NewAzureRmAks.cs

```diff
@@ -72,6 +82,76 @@ private void PreValidate()
             if ((this.IsParameterBound(c => c.NodeMinCount) || this.IsParameterBound(c => c.NodeMaxCount) || this.EnableNodeAutoScaling.IsPresent) &&
                 !(this.IsParameterBound(c => c.NodeMinCount) && this.IsParameterBound(c => c.NodeMaxCount) && this.EnableNodeAutoScaling.IsPresent))
                 throw new PSInvalidCastException(Resources.AksNodePoolAutoScalingParametersMustAppearTogether);
+
+            if (this.IsParameterBound(c => c.GenerateSshKey) && this.IsParameterBound(c => c.SshKeyValue))
+            {
+                throw new ArgumentException(string.Format(Resources.DonotUseGenerateSshKeyWithSshKeyValue));
+            }
+        }
+
+        private void VerifySshKeyGenBinaryExist()
+        {
+            using (Process process = new Process())
+            {
+                if ((RuntimeInformation.IsOSPlatform(OSPlatform.Windows)))
+                {
+                    process.StartInfo.FileName = "where.exe";
+                }
+                else
+                {
+                    process.StartInfo.FileName = "whereis";
+                }
+                process.StartInfo.Arguments = "ssh-keygen";
+                process.StartInfo.UseShellExecute = false;
+                process.StartInfo.RedirectStandardOutput = true;
+
+                process.Start();
+                process.WaitForExit();
+
+                string result = process.StandardOutput.ReadLine();
+                if (result.Contains("not found") || result.Contains("Could not find") || result.Trim().Equals("ssh-keygen:"))
+                {
+                    throw new ArgumentException(Resources.EnableSsh);
+                }
+
+                if (process.ExitCode != 0)
+                {
+                    throw new ArgumentException(Resources.EnableSsh);
+                }
+            }
+        }
+
+        private string GenerateSshKeyValue()
+        {
+            VerifySshKeyGenBinaryExist();
+            String generateSshKeyPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".ssh", "id_rsa"); ;
+            if (File.Exists(generateSshKeyPath))
+            {
+                throw new ArgumentException(string.Format(Resources.DefaultSshKeyAlreadyExist));
+            }
+            using (Process process = new Process())
+            {
+                process.StartInfo.FileName = "ssh-keygen";
+                process.StartInfo.Arguments = "-f " + generateSshKeyPath;
```

Comment:
It is possible that path contains white space, please add double quote to surround the path.